### DISCUSSION TITLE
Why update and updateMany action generate two query?

### DISCUSSION THREAD

**Discussion by notzheng** [prisma/prisma#5847] [Q&A] [answered]
for example, when I run code follows:

```javascript

const result = await this.prismaService.good.update({
      where: {
        id: 1,
      },
      data: {
        stock: {
          decrement: 1,
        },
      },
    });

```

I fount prisma generate two queries( exclude the last select query) in a transaction, I want to know why not use only one query to update ?

```sql

prisma:query BEGIN
1⃣️ prisma:query SELECT `whms`.`goods`.`id` FROM `whms`.`goods` WHERE `whms`.`goods`.`id` = ?
2⃣️ prisma:query UPDATE `whms`.`goods` SET `stock` = (`stock` - ?), `updated_at` = ? WHERE `whms`.`goods`.`id` IN (?)
prisma:query SELECT `whms`.`goods`.`id`, `whms`.`goods`.`name`, `whms`.`goods`.`quality`, `whms`.`goods`.`stock`, `whms`.`goods`.`created_at`, `whms`.`goods`.`updated_at` FROM `whms`.`goods` WHERE `whms`.`goods`.`id` = ? LIMIT ? OFFSET ?
prisma:query COMMIT

```

I think it should be 

```sql
prisma:query BEGIN
1⃣️ prisma:query UPDATE `whms`.`goods` SET `stock` = (`stock` - ?), `updated_at` = ? WHERE `whms`.`goods`.`id` = (?)
prisma:query SELECT `whms`.`goods`.`id`, `whms`.`goods`.`name`, `whms`.`goods`.`quality`, `whms`.`goods`.`stock`, `whms`.`goods`.`created_at`, `whms`.`goods`.`updated_at` FROM `whms`.`goods` WHERE `whms`.`goods`.`id` = ? LIMIT ? OFFSET ?
prisma:query COMMIT

```

**Comment by nikolasburk** [comment, marked as answer]
Hey @notzheng, thanks a lot for raising this! There are various factors that determine what kind of queries Prisma Client generates. In this case, it might be that Prisma Client doesn't generate the most efficient SQL query. If you'd like to see this improved, can you please [create a bug report](https://github.com/prisma/prisma/issues/new/choose) so that our engineering team can start looking into potentially optimizing this query? 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
